libraries/RW/K8s/namespace_tasks_mixin.py

```diff
@@ -334,4 +335,176 @@ def count_notready_pods (
+
+        Args: 
+            :namespace str: The namespace to query for results.  
+            :context str: The kubnerets context to use, as listed in the kubeconfig secret. 
+            :custom_resources list: A list of custom resources to search for. 
+            :kubeconfig paltform.Secret: A kubeconfig that provides access to the necessary resources. 
+            :target_service platform.Service: Which service to use (typically kubectl) 
+            :binary_name str:  The binary to use. Typically kubectl, but could also be oc, or another k8s distribution.
+            :return str: The string output of the query results.  
+
+        """
+        ## TODO add filtering / search capability
+        output: str = ""
+        for resource in custom_resources: 
+          cmd = f"{binary_name} describe {resource} -n {namespace} --context {context}" 
+          crd_list: str = self.shell(
+              cmd=cmd,
+              target_service=target_service,
+              kubeconfig=kubeconfig,
+          )
+          output=crd_list + output
+        return output
+
+    def fetch_pod_logs_and_events_by_label (
+        self, 
+        namespace:str,
+        context:str,
+        kubeconfig: platform.Secret,
+        target_service: platform.Service,
+        binary_name: str = "kubectl",
+        resource_labels: str = "",
+        log_lines: int=100
+    ) -> str:
+        """Takes in a list of labels, a namespace, and a context and returns the logs and events for each pod that matches the label. 
+
+        Args: 
+            :namespace str: The namespace to query for results.  
+            :context str: The kubnerets context to use, as listed in the kubeconfig secret. 
+            :custom_resources list: A list of custom resources to search for. 
+            :kubeconfig paltform.Secret: A kubeconfig that provides access to the necessary resources. 
+            :target_service platform.Service: Which service to use (typically kubectl) 
+            :binary_name str:  The binary to use. Typically kubectl, but could also be oc, or another k8s distribution.
+            :resources_labels str: A list of labels to select pods with. 
+            :log lines int: The number of log lines to include for each pod. Defaults to 100, -1 includes all pod logs. 
+            :return str: The stdout of the query as a string.  
+
+        """
+        output: str = ""
+        pod_names=self.fetch_pod_names_by_label(namespace, context, kubeconfig, target_service, binary_name, resource_labels)
+        for pod_name in pod_names: 
+          cmd = f"{binary_name} logs {pod_name} --tail={log_lines} --all-containers -n {namespace} --context {context}" 
+          log_output: str = self.shell(
+              cmd=cmd,
+              target_service=target_service,
+              kubeconfig=kubeconfig,
+          )
+          output=f"POD LOGS: {pod_name}\n--------\n{log_output}\n--------\n{output}"
```

Comment:
generating output in a standardized way is tricky. I ran into it a lot when coding "Checks" - we should look at templating output but this is fine for now.

Reply:
Agreed... this was very hacky to just get something readable... I'd like a better report template for sure!